subcommand/discover-servers/command_test.go

```diff
@@ -0,0 +1,104 @@
+package meshinit
+
+import (
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"net/http/httptest"
+	"os"
+	"testing"
+
+	"github.com/aws/aws-sdk-go/service/ecs"
+	"github.com/hashicorp/consul-ecs/awsutil"
+	"github.com/hashicorp/consul-ecs/subcommand/discover-servers/mocks"
+	"github.com/mitchellh/cli"
+	"github.com/stretchr/testify/require"
+)
+
+func TestFlagValidation(t *testing.T) {
+	cases := []struct {
+		args  []string
+		error string
+	}{
+		{
+			nil,
+			"-service-name must be set",
+		},
+		{
+			[]string{"-service-name", "test"},
+			"-out must be set",
+		},
+	}
+
+	ui := cli.NewMockUi()
+	cmd := Command{
+		UI: ui,
+	}
+	for _, c := range cases {
+		code := cmd.Run(c.args)
+		require.Equal(t, code, 1)
+		require.Contains(t, ui.ErrorWriter.String(), c.error)
+	}
+}
+
+func TestDiscoverServers(t *testing.T) {
+	cases := map[string]func(*ecs.Task) []*ecs.Task{
+		"no-tasks": func(_ *ecs.Task) []*ecs.Task {
+			return nil
+		},
+		"no-containers": func(task *ecs.Task) []*ecs.Task {
+			task.Containers = nil
+			return []*ecs.Task{task}
+		},
+		"no-network-interfaces": func(task *ecs.Task) []*ecs.Task {
+			task.Containers[0].NetworkInterfaces = nil
+			return []*ecs.Task{task}
+		},
```

Comment:
should we also test for a case when there is no `PrivateIpv4Address`?

Reply:
Yep, added! Also, added a case where containers are returned, but none are named `consul-server`.